MainWindow.xaml.cs

```diff
@@ -113,7 +112,7 @@ private void tbr_Options_Click(object sender, RoutedEventArgs e)
             // lazy initialization
             if (_optionsWindow == null)
             {
-                _optionsWindow = new OptionsWindow(_options);
+                _optionsWindow = new OptionsWindow(this, _options);
```

Comment:
Clarification: Why is `this` needed here now? Is that the parenting aspect?

Reply:
Because `OptionsWindow` calls `MainWindow.OnOptionsUpdated` so that we can adjust things that should change (eg `GetOutputNames()`)